scipy/linalg/_matfuncs.py

```diff
@@ -547,6 +545,7 @@ def funm(A, func, disp=True):
     .. [1] Gene H. Golub, Charles F. van Loan, Matrix Computations 4th ed.
 
     """
+    from ._multimethods import schur, rsf2csf
```

Comment:
Why this import is not at the top of the file?

Reply:
That is because of the circular import issue. I'm was not sure whether we should import `schur, rsf2csf` from `multimethods` or `_decomp_schur`. I think importing from `_decomp_schur` should also work fine with all backends, which can make this patch more clear.

Reply:
I think this should be removed. Once we're inside a function, we don't want to be calling the public `linalg.xxx` functions anymore (which again check for dispatchables), but rather the scipy-internal ones below the uarray layer. Because at this point we already know we're not dispatching elsewhere.

Reply:
Yep, agreed!